src/lib.rs

```diff
@@ -71,7 +71,46 @@ impl BigInt<[Digit; 1]> {
     }
 }
 
-impl<D1, D2> std::ops::Add<&BigInt<D2>> for &BigInt<D1>
+macro_rules! impl_digit_operation {
+    ($name:ident,$op:ident,$tp:ty) => {
```

Comment:
I'd give more clear names:
```suggestion
    ($op_trait:ident, $op:ident, $rhs:ty) => {
```

Reply:
One more stylistic suggestion: let's combine all similar macro calls?

Ie instead of writing:

```rust
 impl_digit_operation!(Add, add, u8);
 impl_digit_operation!(Add, add, u16);
 impl_digit_operation!(Add, add, u32);
 impl_digit_operation!(Mul, mul, u8);
 impl_digit_operation!(Mul, mul, u16);
 impl_digit_operation!(Mul, mul, u32);
 // ...
```

it could be better to combine all of them and replace with:

```rust
impl_digit_operation! {
    (Add, add, u8),
    (Add, add, u16),
    (Add, add, u32),
    (Mul, mul, u8),
    (Mul, mul, u16),
    (Mul, mul, u32),
    // ...
}
```

In this particular macro, it can be done by using repeating template `$($op_trait:ident, $op:ident, $rhs:ty),+`.